app/scripts/controllers/charts.js

```diff
@@ -93,22 +94,18 @@ app.controller('ChartsCtrl', ['$scope', '$modal', 'ConfigService', 'ForamAPIServ
   };
 
   var generateChart = function () {
-    if ($scope.chartParams.gene1 || $scope.chartParams.gene2) {
+    if ($scope.chartParams.gene) {
 
-      var gene1 = $scope.chartParams.gene1.replace(/\s+/g, '');
+      var gene = $scope.chartParams.gene.replace(/\s+/g, '');
 
       var flatParams = {
         start: $scope.chartParams.start,
         stop: $scope.chartParams.stop,
         "genes[]": [
-          gene1
-        ]
+          gene
+        ],
+        "group_by": $scope.chartParams.groupingParameter
```

Comment:
Rozumiem, że ta dziwaczna notacja `"genes[]": [...]` wynika z dziwnego formatu wymaganego przez API? Jeżeli byłby czas to może warto by to jakoś uprościć. W każdym razie, `"group_by": value` jest równoznaczne `group_by: value`, więc przynajmniej stąd można wywalić ten cudzysłów :)